### ISSUE TITLE
Unable to append a content line that starts with a dash

### ISSUE THREAD

**Issue by jkaiser-ioki** [xwmx/nb#204]
Appending content with a leading dash to existing notes via the edit command does not work as expected.

The following happens

```shell
[jonas@orchid ~]$ nb a new-test.md --title "Reproduce Dash Bug" --content "first line"
grep: warning: stray \ before @
grep: warning: stray \ before !
grep: warning: stray \ before /
Added: [5] new-test.md "Reproduce Dash Bug"
```

(We can probably ignore the grep warnings, but I opened another issue for this, just in case)

The following works:
```shell
[jonas@orchid ~]$ nb edit 5 --content "plain append"
Updated: [5] new-test.md "Reproduce Dash Bug"
```

this one doesn't (i get the default help/usage blurb [truncated for readability])
```shell
[jonas@orchid ~]$ nb edit 5 --content "- dash append"
 __          _
 \ \   _ __ | |__
  \ \ | '_ \| '_ \
  / / | | | | |_) |
 /_/  |_| |_|_.__/

 ------------------

[nb] Command line and local web note-taking, bookmarking, and archiving with
plain text data storage, encryption, filtering and search, pinning, #tagging,
Git-backed versioning and syncing, Pandoc-backed conversion, global and local
notebooks, customizable color themes, [[wiki-style linking]], plugins, and more
in a single portable, user-friendly script.

Help:
  nb help               Display this help information.
  nb help <subcommand>  View help information for <subcommand>.
  nb help --colors      View information about color settings.
  nb help --readme      View the `nb` README file.

Usage:
  nb
  nb [<ls-options>...] [<id> | <filename> | <path> | <title> | <notebook>]
  nb [<url>] [<bookmark options>...]
  ...
```

if I do the same thing on a note that is nested inside folders, it also does not work but fails in a different way, i.e. i get the following:
```shell
[jonas@orchid ~]$ nb edit 4/1/1/1 --content "- some content after a leading dash"
__          _
\ \   _ __ | |__
 \ \ | '_ \| '_ \
 / / | | | | |_) |
/_/  |_| |_|_.__/
------------------
nb shell started. Enter ls to list notes and notebooks.
Enter help for usage information. Enter exit to exit.
! Not found: --content
```

This feels as if something is not properly escaped and accidentally parsed as a control character, rather than actual content

Am I missing something obvious here?

**Comment by xwmx** [comment]
Hi @jkaiser-ioki, this should be resolved as of version 7+. Let me know if you run into additional issues with it. Thanks!

**Comment by jkaiser-ioki** [comment]
I don't think this has been fixed ... the latest version (7.1.2) no longer spews out the usage page but it also does not append the provided `--content` to the note.

The use case I have in mind for this:

1. create a markdown note file and write up some header information
2. while I work on something (usually some kind of research task where I want to jot down my thoughts along the way), repeatedly fire shell commands that append details I want to persist. Each of those items should be appended with a leading dash inside the note to generate a nice markdown bullet list (to this end, an option to append without a blank line would of course also be a nice feature ;-) )

before we go into depth ... does my use case even make sense here, and/or is there a better mechanism for this? Also, can I bypass the whole problem by quoting my content differently?

**Comment by xwmx** [comment]
@jkaiser-ioki It should work now in the latest version in the repo. I added experimental support for `edit --append <content>` and `edit --prepend <content>` that adds the content without the extra newline. When the `--content <content>` option is used, the extra newline is still included.
